feature(Management List): A new design pattern for sorting/filtering items #260

richavyas opened the pull request:
## Description

Management List, a new design pattern for sorting/filtering items. Takes care of #260 

### What's included?
- Management List with sorting & filtering design pattern. 
- Filter options to be added as one of the column headers and the list can be filtered based on what option has been selected. 
- Based on the selected filter option, list can also be sorted say `asc/desc`.

#### Test Steps

- [x] ng serve
- [x] then navigate to http://localhost:4200/#/style-guide/management-list
- [x] 🎉 

![filter-sort-menu](https://cloud.githubusercontent.com/assets/4469548/22004554/565370e2-dc11-11e6-8ae5-e504a3abfbb8.gif)

kyleledbetter commented:
I'm not sure if design patterns and non-platform code needs unit tests

emoralesb05 commented on src/platform/core/management-list/management-list.module.ts:
```diff
@@ -0,0 +1,28 @@
+    TdManagementListComponent,
+  ],
+})
+export class CovalentManagementListModule {
```
Im not sure we want to add a module just for a design pattern. The whole code should be in the design pattern page.

emoralesb05 commented on src/platform/core/management-list/management-list.ts:
```diff
@@ -0,0 +1,41 @@
+  styleUrls: ['./management-list.scss'],
+})
+
+export class TdManagementListComponent {
```
Any reason this is a component and not a service?

emoralesb05 commented on src/app/components/style-guide/management-list/management-list.component.html:
```diff
@@ -0,0 +1,94 @@
+            <h3 md-line>App Name</h3>
+            <span flex></span>
+            <span flex="35" class="pad-left md-body-1 tc-grey-500">Owner</span>
+            <span class="md-sort-header" flex="40" layout="row" layout-align="center center" class="pad-left md-body-1 tc-grey-500">
```
We can probably leverage the `md-select` now to do this.

emoralesb05 commented on src/app/components/style-guide/management-list/management-list.component.ts:
```diff
@@ -0,0 +1,64 @@
+    created_at: OrderBy.ASC,
+  };
+
+  constructor(@Inject(forwardRef(() => TdManagementListComponent)) private _parent: TdManagementListComponent) {
```
For a `design pattern` purpose, i think its better to just have the `sortBy` function here.

I get why we would want to reuse it, but we always want to push people to sort in the server, and not in the client.

kyleledbetter commented:
@richavyas this looks fantastic. As discussed let's keep the pattern to just mock & pattern, and update the spec in the issue https://github.com/Teradata/covalent/issues/260

So i'd recommend moving any logic the platform for now and keep this limited to design pattern in the docs app

richavyas commented:
- [x] Changed this to pattern & mock
- [x] replaced menu to `select` to load filter options

![filter-sort-select](https://cloud.githubusercontent.com/assets/4469548/22079932/64e5b252-dd72-11e6-845e-16c6d9591676.gif)

kyleledbetter commented:
Pushed updates:
![manage-list-pattern](https://cloud.githubusercontent.com/assets/867883/22083265/d8620b7c-dd90-11e6-9258-a8658ca32bc5.gif)

emoralesb05 commented:
Pushed a couple of tweaks to leverage the usage of `md-select` and make the example more generic
